### PR TITLE
elastic search by criteria

### LINKED ISSUE #45: search autocomplete feature with pagination

### PR DESCRIPTION
**Pull request by eyakauleva**
close #45

### CODE REVIEW

**Review by h1alexbel — COMMENTED** [review]
nice, @eyakauleva see my comments

File: src/main/java/com/solvd/micro9/users/domain/elasticsearch/ElstcUser.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.data.annotation.Id;
+import org.springframework.data.elasticsearch.annotations.Document;
+import org.springframework.data.elasticsearch.annotations.Field;
+import org.springframework.data.elasticsearch.annotations.FieldType;
+
+@Document(indexName = "syncmongoelastic")
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class ElstcUser {
```
[the remark this answers is not in the document]
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/users/domain/elasticsearch/ElstcUser.java:17]: @h1alexbel it's like ElasticUser but shortened, is it bad?

File: src/main/java/com/solvd/micro9/users/persistence/elastic/ReactiveElasticFilterImpl.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.elasticsearch.core.ReactiveElasticsearchOperations;
+import org.springframework.data.elasticsearch.core.SearchHit;
+import org.springframework.data.elasticsearch.core.query.Query;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Flux;
+
+@Component
+@RequiredArgsConstructor
+public class ReactiveElasticFilterImpl implements ReactiveElasticFilter {
```
[the remark this answers is not in the document]
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/users/persistence/elastic/ReactiveElasticFilterImpl.java:16]: @h1alexbel can I leave it just as ElasticFilter?

[the remark this answers is not in the document]
Comment (h1alexbel) [reply src/main/java/com/solvd/micro9/users/persistence/elastic/ReactiveElasticFilterImpl.java:16]: Sounds OK, but how does this name (`ElasticFilter`) denote the object being ReactiveElasticFilter, so implementation class name should be understandable why and what is it

File: src/main/java/com/solvd/micro9/users/domain/elasticsearch/ElstcUser.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (h1alexbel) [reply src/main/java/com/solvd/micro9/users/domain/elasticsearch/ElstcUser.java:17]: @eyakauleva in my opinion, `EsUser` will fit better

File: src/main/java/com/solvd/micro9/users/persistence/elastic/ReactiveElasticFilterImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/users/persistence/elastic/ReactiveElasticFilterImpl.java:16]: @h1alexbel I will rename them both

File: src/main/java/com/solvd/micro9/users/domain/elasticsearch/ElstcUser.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/users/domain/elasticsearch/ElstcUser.java:17]: @h1alexbel that's the problem. I already have EsUser that stands for EventSourcingUser

[the remark this answers is not in the document]
Comment (h1alexbel) [reply src/main/java/com/solvd/micro9/users/domain/elasticsearch/ElstcUser.java:17]: @eyakauleva yeah, I remember

try to stay away from long names as much as possible

File: src/main/java/com/solvd/micro9/users/persistence/elastic/ReactiveElasticFilterImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/users/persistence/elastic/ReactiveElasticFilterImpl.java]: fixed acce3e8

**Review by h1alexbel — APPROVED** [review]
@eyakauleva good one

**Pull request merged**